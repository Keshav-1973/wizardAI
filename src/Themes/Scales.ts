export enum Themes {
    DARK_MODE = 'DARK_MODE',
    LIGHT_MODE = 'LIGHT_MODE',
    HIGH_CONTRAST = 'HIGH_CONTRAST',
}

export enum Spacings {
    XL = 'xl',
    L = 'l',
    M = 'm',
    S = 's',
    XS = 'xs',
}

export enum ColorPalette {
    BLUE = '',
    RED = '#B92406',
    BLACK = '#0B0B0B',
    WHITE = '#FFFFFF',
    DARK_BLUE = '#002852',
    BLACK_BUTTON = '#000000',
    ACTIVE_WHITE = '#F7F9FA',
    ACTIVE_BLACK = '#1F1F1F',
    GREY = '#595959',
    // DARK_GREEN = "#40744D",
    INPUT_LIGHT = "#dae5dd",
    INPUT_DARK = "#545252",
    DARK_GREEN = "#008000",

}

export enum SemanticColors {
    MAIN_BACKGROUND = 'mainBackground',
    MAIN_FOREGROUND = 'mainForeground',
    PRIMARY_BUTTON = 'primaryButton',
    SECONDARY_BUTTON = 'secondaryButton',
    PRIMARY_BUTTON_COLOR = 'darkBlue',
    ERROR_COLOR = 'error',
    ACTIVE_TAB = 'activeTabBackground',
    SECONDARY_TEXT = 'secondaryText',
    HEADING = 'heading',
    INPUT_FIELD = "inputField",
    SUBSCRIPT = "subscript",
    SECONDARY_BUTTON_TEXT="secondaryButtonText",
    PRIMARY_BUTTON_TEXT="primaryButtonText"
}

export enum SupportedDevices {
    PHONE = 'PHONE',
    TABLET = 'TABLET',
    DESKTOP = 'DESKTOP',
}
