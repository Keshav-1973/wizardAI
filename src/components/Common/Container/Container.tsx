import React from 'react';
import {StyleSheet, StyleProp, ViewStyle} from 'react-native';
import {SemanticColors} from '@screens/../Themes/Scales';
import ViewComponent from '@components/Common/ViewComponent/ViewComponent';

type Props = {
  children: JSX.Element | JSX.Element[];
  style?: StyleProp<ViewStyle>;
  isPadded?: boolean;
};

const Container = ({children, style, isPadded}: Props) => {
  return (
    <ViewComponent
      style={[styles(isPadded).wrapper, style]}
      backgroundColor={SemanticColors?.MAIN_BACKGROUND}>
      {children}
    </ViewComponent>
  );
};

export default Container;

const styles = (isPadded: boolean) =>
  StyleSheet.create({
    wrapper: {
      padding: isPadded ? 10 : 0,
      flex: 1,
      height: '100%',
    },
  });
