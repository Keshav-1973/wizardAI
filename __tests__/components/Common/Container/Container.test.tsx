import React from 'react';
import {render} from '@testing-library/react-native';
import Container from '@components/Common/Container/Container';
import {ColorPalette} from '@themes/Scales';

const mockTheme = {
  colors: {
    MAIN_BACKGROUND: ColorPalette.WHITE,
  },
};

jest.mock('@screens/../Themes/Scales', () => ({
  ...jest.requireActual('@screens/../Themes/Scales'),
  SemanticColors: mockTheme?.colors,
}));

describe('<Container /> snapshot', () => {
  it('matches the snapshot', () => {
    const {toJSON} = render(
      <Container>
        <></>
      </Container>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
