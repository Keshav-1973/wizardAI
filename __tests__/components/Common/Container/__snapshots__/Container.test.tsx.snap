// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> snapshot matches the snapshot 1`] = `
<View
  style={
    [
      {},
      [
        {
          "flex": 1,
          "height": "100%",
          "padding": 0,
        },
        undefined,
      ],
    ]
  }
/>
`;
